src/Controller/InvalidateTokenController.php

```diff
@@ -66,25 +68,27 @@ public function tokenAction(Request $request)
     {
         $response = new Response();
 
-        if (!$this->isFromTrustedProxy($request)) {
-            $response->setStatusCode('401', 'Unauthorized');
+        // Important to keep this condition, as .vcl rely on this to prevent everyone from being able to fetch the token.
+        if ($request->headers->get('accept') !== 'application/vnd.ezplatform.invalidate-token') {
+            $response->setStatusCode(400, 'Bad request');
 
             return $response;
         }
 
-        // Important to keep this condition, as .vcl rely on this to prevent everyone from being able to fetch the token.
-        if ($request->headers->get('accept') !== 'application/vnd.ezplatform.invalidate-token') {
-            $response->setStatusCode('400', 'Bad request');
+        $this->tagHandler->addTags(['ez-invalidate-token']);
+
+        $token = $this->configResolver->getParameter('http_cache.varnish_invalidate_token');
+        if ($request->headers->get(self::TOKEN_HEADER_NAME) !== $token) {
```

Comment:
I don't get this...
so, the trusted proxy varnish does not know the invalidate token, so it will try to [retrieve it](https://github.com/ezsystems/ezplatform-http-cache/blob/master/docs/varnish/vcl/varnish5.vcl#L234) by accessing `/_ez_http_invalidatetoken`. But now that is rejected because it doesn't have the invalidation token....

Or am I missing something?

Reply:
ah... It will use the token it received in the initial PURGE request

Reply:
Yes, it's my favorite part of this PR)

Reply:
done